core/kernel/impl/class.ApiModelOO.php

```diff
@@ -47,22 +46,43 @@ class core_kernel_impl_ApiModelOO
     extends core_kernel_impl_Api
         implements core_kernel_api_ApiModel
 {
-    // --- ASSOCIATIONS ---
+    /**
+     * @var core_kernel_impl_ApiModelOO
+     */
+    private static $instance = null;
 
+    /** @var ModelFactory */
+    private $modelFactory;
 
-    // --- ATTRIBUTES ---
+    /**
+     * @author firstname and lastname of author, <author@example.org>
+     * @return void
+     */
+    private function __construct()
+    {
+        $serviceManager = ServiceManager::getServiceManager();
+        $this->modelFactory = $serviceManager->get(ModelFactory::SERVICE_ID);
+    }
 
     /**
-     * Short description of attribute instance
-     *
-     * @access private
-     * @var ApiModelOO
+     * @author firstname and lastname of author, <author@example.org>
+     * @return core_kernel_impl_ApiModelOO
      */
-    private static $instance = null;
+    public static function singleton()
+    {
+        $returnValue = null;

```

Comment:
normally 
```
    if (self::$instance == null)
    {
      self::$instance = new Singleton();
    }
 
    return self::$instance;
```
but your implementation is different, can you please comment on the value of this form